flink-shaded-zookeeper-parent/pom.xml

```diff
@@ -85,6 +85,12 @@ under the License.
                     </execution>
                 </executions>
             </plugin>
+
+            <plugin>
+                <!-- Used to resolve variables in the 'version' tag -->
+                <groupId>org.codehaus.mojo</groupId>
+                <artifactId>flatten-maven-plugin</artifactId>
```

Comment:
The `build` section defines which plugins are actually run (in contrast to the `pluginManagement` section in the root pom, which more or less serves as a template). This `build` section is inherited by sub modules, just like the shade-plugin configuration., and hence doesn't need to be added explicitly to the sub-modules.

Reply:
I was asking because this is not how it is done for other parent/submodules, e.g. `flink-shaded-hadoop-2`

Reply:
The execution of the flatten plugin must be defined below the shade-plugin execution, to ensure the flattening is executed after the shade-plugin. If it runs before the shade-plugin it has no effect.

The zookeeper parent is the only parent module defining the shade-plugin configuration, since the shading logic is indeed identical in both child modules. This is not the case for hadoop for example, hence the shade-plugin execution is additionally listed in the sub-modules, hence they also have to list the flatten plugin to ensure the execution order.

Reply:
Alright, thanks for the explanation